### PR TITLE
Add highly requested Dark Mode and Logo support

### PR DESCRIPTION
**Pull request by barry-smithjr**
# This PR enables dark mode and logo support

## Technical Details

-  data-bs-theme HTML element is set to "auto"
- When page loads, the JS reads the dark/light preference from the browser and sets accordingly on page and localstorage.
- Toggle in the header will update the localstorage and change the page theme.
- `site_logo` setting added to add a logo.
- `display_title` setting added to enable/disable showing of `site_title` next to the logo.

## Changes summary:

- Removed hardcoded white backgrouds from textareas
- Changed footer to use an included color for readability
- Added a header that uses the site_title setting to display app name
- Added light/dark toggle 
- Included Font Awesome 6 solid icons for the light dark toggle
- Bootstrap upgraded to 5.3.3
- Announcement banner moved under header
- Moved "click to copy" code to individual pages

## Screenshots

flashpaper-dark
flashpaper-dark-announcement
flashpaper-light
flashpaper-light-announcement

### CODE REVIEW

File: settings.example.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 	$settings = [
 		'site_title' => 'FlashPaper :: Self-Destructing Message',
+		'site_logo' => 'img/logo.png',
+		'display_title' => 'false', # true/false
```
Comment (AndrewPaglusch) [inline settings.example.php:7]: It looks like the two new settings you added might be missing from the `docker/entrypoint.sh` and `docker/settings.php.TEMPLATE` files, so people who use Docker won't get the new settings

File: js/color-toggle.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*!
+ * Originally from: https://mdbootstrap.com/docs/standard/content-styles/theme/
+ * with modifications to support Bootstrap
+ */
+
+const themeStitcher = document.getElementById("themingSwitcher");
```
Comment (AndrewPaglusch) [inline js/color-toggle.js:6]: Possible typo?

File: html/header.php [hunk 3]
```diff
@@ -69,20 +70,46 @@
...
+					<span class="navbar-text">
+						<li class="nav-item align-items-center d-flex" >
+ 							<i class="fa-solid fa-sun"></i>
+				  			<!-- Default switch -->
+  							<div class="ms-2 form-check form-switch">
+    						<input class="form-check-input" type="checkbox" role="switch" id="themingSwitcher" />
+  							</div>
+  							<i class="fa-solid fa-moon"></i>
+						</li>
+		      		</span>
```
Comment (AndrewPaglusch) [inline html/header.php:107]: Incorrect indentation

File: html/header.php [hunk 4]
```diff
@@ -69,20 +70,46 @@
...
-		<script>
+	<script>
 			if (location.protocol != 'https:') {
 				document.write('<div style="padding-top: 1%" class="container"><div class="alert alert-danger"><strong>Danger!</strong> This site is not being accessed over an encrypted connection. Do NOT input any sensitive information!</div></div>');
 			}
-			function copyText() {
-				var textToCopy = document.getElementById("copy");
-				textToCopy.select();
-				document.execCommand("copy");
-			}
```
Comment (AndrewPaglusch) [inline html/header.php:82]: Any reason we're adding this same block to multiple pages instead of just once?

File: html/header.php [hunk 5]
```diff
@@ -69,20 +70,46 @@
...
+							if ( $settings['site_logo'] != '' ) {
+								echo '<img src="'. $settings['site_logo'] .'" alt="Logo" width="200" class="d-inline-block align-middle">';
+							}
+						?>
+						<?php
+							if ( $settings['display_title'] == 'true') {
+								echo '' . $settings['site_title'] . '';
+							} else {
+								echo '';
+							}
```
Comment (AndrewPaglusch) [inline html/header.php:95]: Why are we echoing out empty strings here? Can we do something like this instead?

```
if ( $settings['display_title'] == 'true') {
    echo $settings['site_title'];
}
```

**Review by AndrewPaglusch — CHANGES_REQUESTED** [review]
Just some small changes and questions @barry-smithjr. Thanks!

File: html/header.php (hunk 5 above)
Comment (barry-smithjr) [reply html/header.php]: fixed in [3ce241d](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/3ce241dd1de608ab90fad606d6ad1d0ef2adaf5a)

File: html/header.php (hunk 4 above)
Comment (barry-smithjr) [reply html/header.php]: This was removed and added to the individual pages after adding the theme toggle. After doing some debugging in html/header.php, I discovered a missing closing script tag. All fixed up in [8171400](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/8171400912bc3ba6b3135d2e04231c90ee80b1bb)

File: html/header.php (hunk 3 above)
Comment (barry-smithjr) [reply html/header.php:110]: Corrected in [8171400](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/8171400912bc3ba6b3135d2e04231c90ee80b1bb)

File: js/color-toggle.js (hunk 2 above)
Comment (barry-smithjr) [reply js/color-toggle.js]: It was a typo, fixed in [8171400](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/8171400912bc3ba6b3135d2e04231c90ee80b1bb)

File: settings.example.php (hunk 1 above)
Comment (barry-smithjr) [reply settings.example.php:7]: Added settings to both files in [8171400](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/8171400912bc3ba6b3135d2e04231c90ee80b1bb)

Updated docker/entrypoint.sh to no logo when no url is supplied in [713b3f5](https://github.com/AndrewPaglusch/FlashPaper/pull/159/commits/713b3f5569fae38b6cc3b70f5b682e79b5ac89ec)

**Review by mattburchett — APPROVED** [review]

**Review by AndrewPaglusch — APPROVED** [review]

**Pull request merged**